packages/@sfx/sayt-driver-plugin/test/unit/browser/sayt-driver-plugin.test.ts

```diff
@@ -0,0 +1,193 @@
+
+      expect(Driver.core).to.deep.equal(Plugins);
+    });
+  });
+
+  describe('.ready()', () => {
+    let registerListener: any;
+
+    beforeEach(() => {
+      registerListener = spy(Driver.core['dom_events'], 'registerListener');
+    });
+
+    it('should register an event listener for receiving sayt API requests', () => {
+      Driver.ready();
+
+      registerListener.getCall(0).calledWith(Driver.saytDataEvent, Driver.fetchSaytData);
+    });
+  });
+
+  describe('.unregister()', () => {
+    let unregisterListener;
+
+    beforeEach(() => {
+      unregisterListener = spy(Driver.core['dom_events'], 'unregisterListener');
+    });
+
+    it('should unregister the sayt event listener', () => {
+      Driver.unregister();
+
+      expect(unregisterListener).to.have.been.calledWith(Driver.saytDataEvent, Driver.fetchSaytData);
+    });
+  });
+
+  describe('.autocompleteCallback', () => {
+    it('should return an array of search term strings from the sayt response', () => {
+      const response = {
+        result: {
+          searchTerms: [
+            { value: 'a' , notValue: 'z' },
+            { value: 'b' },
+            { value: 'c' },
+          ],
+        },
+      };
+
+      const cbReturn = Driver.autocompleteCallback(undefined, response);
+
+      expect(cbReturn).to.deep.equal(['a', 'b', 'c']);
+    });
+
+    it('should only return entries that have a searchterm value', () => {
+      const response = {
+        result: {
+          searchTerms: [
+            { value: 'a' },
+            {},
+            { value: 'c' },
+          ],
+        },
+      };
+
+      const cbReturn = Driver.autocompleteCallback(undefined, response);
+
+      expect(cbReturn).to.deep.equal(['a', 'c']);
+    });
+
+    it('should return an empty array if there are no searchTerms', () => {
+      const response = {
+        result: {
+          searchTerms: [],
+        },
+      };
+
+      const cbReturn = Driver.autocompleteCallback(undefined, response);
+
+      expect(cbReturn).to.deep.equal([]);
+    });
+  });
+
+  describe('.sendSaytAPIRequest()', () => {
+    const saytDataPayload = {
+      query: 'shirt',
+      collection: 'backup',
+    };
+    let autocompleteCallback;
+
+    beforeEach(() => {
+      autocompleteCallback = stub(Driver, 'autocompleteCallback');
+    });
+
+    it('should make a search call through the sayt client', () => {
+      const autocomplete = spy(Driver.core.sayt, 'autocomplete');
+
+      Driver.sendSaytAPIRequest(saytDataPayload);
+
+      expect(autocomplete).to.be.calledWith(
+        saytDataPayload.query,
+        { collection: 'backup' },
+        autocompleteCallback,
+      );
+    });
+
+    it('should return the result of the Sayt API callback', () => {
+      autocompleteCallback.returns(['a', 'b']);
+      const callbackReturn = autocompleteCallback();
+
+      const returnValue = Driver.sendSaytAPIRequest(saytDataPayload);
+
+      expect(returnValue).to.eventually.equal(callbackReturn);
```

Comment:
Assertions using chai-as-promised should be returned so that mocha can wait on it.

Reply:
@geoffcoutts, you could try skipping this test to see if the assertion is actually included in the final total.

Reply:
Skipping the test wouldn't have an effect. This test passes because no exceptions are thrown during its runtime. A better way to check is to resolve to something that's explicitly not the expected return value and see if it fails.